apps/explorer/src/pages/validator/ValidatorDetails.tsx

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { is, SuiObject, Base64DataBuffer } from '@mysten/sui.js';
+import { useMemo } from 'react';
+import { useParams, Navigate } from 'react-router-dom';
+
+import ErrorResult from '~/components/error-result/ErrorResult';
+import { StatsCoin } from '~/components/top-validators-card/StatsCoin';
+import { useGetObject } from '~/hooks/useGetObject';
+import {
+    VALIDATORS_OBJECT_ID,
+    type ValidatorState,
+} from '~/pages/validator/ValidatorDataTypes';
+import { Card } from '~/ui/Card';
+import { DescriptionList, DescriptionItem } from '~/ui/DescriptionList';
+import { Heading } from '~/ui/Heading';
+import { ImageIcon } from '~/ui/ImageIcon';
+import { AddressLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Stats } from '~/ui/Stats';
+import { Text } from '~/ui/Text';
+import { getName } from '~/utils/getName';
+import { getStakedPercent } from '~/utils/getStakedPercent';
+
+function ValidatorDetails() {
+    const { id } = useParams();
+    const { data, isLoading } = useGetObject(VALIDATORS_OBJECT_ID);
+
+    const validatorsData =
+        data &&
+        is(data.details, SuiObject) &&
+        data.details.data.dataType === 'moveObject'
+            ? (data.details.data.fields as ValidatorState)
+            : null;
+
+    const validatorData = useMemo(() => {
+        if (!validatorsData) return null;
+        return (
+            validatorsData.validators.fields.active_validators.find(
+                (av) => av.fields.metadata.fields.sui_address === id
+            ) || null
+        );
+    }, [id, validatorsData]);
+
+    const validator = useMemo(() => {
```

Comment:
This is basically a view-model, which I don't love. I'd prefer per-field computation for things we need. When we break this down into components more, we can just pass the top-level `validatorData` object and let the component compute it itself.